src/runtime_src/driver/xclng/drm/xocl/subdev/str_qdma.c

```diff
@@ -428,65 +459,101 @@ static ssize_t queue_rw(struct str_device *sdev, struct stream_queue *queue,
 	if (ret) {
 		xocl_err(&sdev->pdev->dev, "Init unmgd buf failed, "
 			"ret=%ld", ret);
-		goto failed;
+		return ret;
 	}
 
 	dir = write ? DMA_TO_DEVICE : DMA_FROM_DEVICE;
 	nents = pci_map_sg(xdev->core.pdev, unmgd.sgt->sgl,
 		unmgd.sgt->orig_nents, dir);
 	if (!nents) {
 		xocl_err(&sdev->pdev->dev, "map sgl failed");
-		ret = -EFAULT;
 		xocl_finish_unmgd(&unmgd);
-		goto failed;
+		return -EFAULT;
 	}
 
-	memset(&wr, 0, sizeof (wr));
-	wr.write = write;
-	wr.len = sz;
-	wr.sgt = unmgd.sgt;
-	wr.eot = (header.flags & XOCL_QDMA_REQ_FLAG_EOT) ? true : false;
-
+	io_req = kmem_cache_alloc(req_cache, GFP_KERNEL);
```

Comment:
Same as previous comment, I think we should avoid kmem_cache_alloc on critical path if possible.

Reply:
Yes, we can pre-alloc 2x of the queue depth at the Q creation time.